sycl/doc/design/IVDEPAndParallelLoopAnnotations.md

```diff
@@ -0,0 +1,195 @@
+```
+
+### Attributes for specifying loop-carried dependence distance: `[[intel::ivdep(safelen)]]` and `[[intel::ivdep(array,safelen)]]`
+
+When placed on a kernel loop, these attributes indicate that any loop-carried
+dependences on all pointer accesses (`ivdep(safelen)`) or on the specified
+arrays/pointers (`ivdep(array,safelen)`) have a distance of at least `safelen`.
+This can be used to disambiguate any aliasing when the compiler cannot
+automatically infer the dependence distance and instead has to make the
+conservative assumption that the distance is 1.
+
+### Marking parallel loops
+
+We are considering adding a mechanism for users to indicate when a kernel loop
+is parallel. THe difference between this and `ivdep` is that a parallel loop
+implies that memory-ordering dependences can be ignored, in addition to memory
+data dependences. Marking parallel loops can either be done using new annotation
+`[[intel::parallel]]`, a new lambda-style `parallel_for` API, or using SYCL
+nested parallelism.
+
+## LLVM Metadata changes
+
+We are proposing the following new LLVM metadata:
+
+1. A new metadata, `llvm.loop.no_depends`, should be generated on loops that are
+   marked with the versions of `ivdep` that don't take a `safelen` (i.e.
+   `intel::ivdep` and `intel::ivdep(array)`). It will replace the
+   previously-generated `llvm.loop.parallel_access_indices` due to the mismatch
+   in the semantics of `llvm.loop.paralllel_accesses` and `ivdep`. The semantics
+   are that all accesses with access groups listed in this metadata have no
+   loop-carried memory data dependences.
+2. A new metadata, `llvm.loop.no_depends_safelen` should be generated on loops
+   that are marked with the versions of ivdep that take a safelen (i.e.,
+   `intel::ivdep(safelen)` and `intel::ivdep(array, safelen)`). It will replace our
+   `llvm.loop.parallel_access_indices_safelen` for the same reason described
+   above. As with `llvm.loop.no_depends`, the semantics indicate that all accesses
+   with access groups listed in this metadata have dependence distances of at
+   least safelen.
+3. The built-in `llvm.loop.parallel_accesses` metadata should be generated on
+   loops that are marked as parallel, or in cases where the compiler knows that
+   the loop is parallel. This metadata allows the compiler to not only ignore
+   loop-carried data dependences, but also loop-carried memory-ordering dependences.
+4. The built-in `llvm.access.group` metadata should be generated for all memory
+   operations (loads, stores, and function calls) in the body of a loop that has
+   either an ivdep attribute, or is marked as parallel (by the user or the
+   compiler). The access group identifier will be passed to the corresponding
+   loop metadata described above.
+
+## SPIR-V Extension
+
+The SPIR-V extension for this design doc is presented in [this sCART
+Review](https://github.com/intel-innersource/documentation.xpu.architecture.spirv-extension-drafts/pull/162)
```

Comment:
That's a private intel repo, so not generally accessible by most viewers of this doc.  Please submit the spec to this repo (under https://github.com/intel/llvm/tree/sycl/sycl/doc/design/spirv-extensions) and link to that copy.